playground/backend/internal/preparators/java_preparators.go

```diff
@@ -137,3 +157,38 @@ func addNewLine(newLine bool, file *os.File) error {
 	}
 	return nil
 }
+
+func changeJavaTestFileName(args ...interface{}) error {
+	filePath := args[0].(string)
+	validationResults := args[1].(*sync.Map)
+	isUnitTest, ok := validationResults.Load(validators.UnitTestValidatorName)
+	if ok && isUnitTest.(bool) {
+		err, className := getPublicClassName(filePath)
+		if err != nil {
+			return err
+		}
+		err = renameJavaFile(filePath, className)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func renameJavaFile(filePath string, className string) error {
+	currentFileName := filepath.Base(filePath)
+	newFilePath := strings.Replace(filePath, currentFileName, fmt.Sprintf("%s%s", className, fs_tool.JavaSourceFileExtension), 1)
+	err := os.Rename(filePath, newFilePath)
+	return err
+}
+
+func getPublicClassName(filePath string) (error, string) {
+	code, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		logger.Errorf("Preparer: Error during open file: %s, err: %s\n", filePath, err.Error())
```

Comment:
Let's keep the consistent naming before the refactoring we planned
```suggestion
		logger.Errorf("Preparator: Error during open file: %s, err: %s\n", filePath, err.Error())
```